### DISCUSSION TITLE
New branch for modernization and automation changes

### DISCUSSION THREAD

**Discussion by danshome** [UCDavisLibrary/ava#918] [General]
Hello @MicheleTobias,

I was wondering if it would be possible to create a new branch from the main one for me. I am making some significant changes and would like to add more automation to the processes. I have been advocating for TTB to modernize their AVA petition processes by accepting GIS features instead of physical maps. With the National Archives now requiring all archives to be digital, I am finally gaining some traction. 

I believe that the UCDavisLibrary AVA project would be an excellent source for TTB to get petitioners' draft AVA boundaries in the form of geojson files. What you have built is an excellent start to making an open-source project for tracking AVAs, and TTB should embrace it. However, I think it needs a little more automation in place. 

I would like a place to submit pull requests for you to consider without destabilizing the main branch. To be clear, I am not suggesting that this project be the source for official AVA data. Instead, I see it as a stepping stone for petitioners, where they could submit drafts of their AVA boundaries for petitions they are working on. It would also be a place where they can get feedback from the open-source community on their AVA boundaries before submitting them to TTB. 

I believe that this project could have a lot of potential for giving TTB an easy way to see what petitioners are working on. It would also give TTB a place where they could give real-time feedback to petitioners and even submit pull requests of their own for any changes they might recommend without having all the back and forth over email and trying to send feature layers back and forth.

What are your thoughts on this?

**Comment by danshome** [comment]
I also have built a sample of an [American Viticulture Area Boundary Builder](https://codepen.io/danshome/pen/YzdjwBd) that would make it easy for petitioners that don't know anything about GIS to create a base geojson file to get them started.

**Comment by kurtishouser** [comment]
@danshome

Since TTB provides Shapefiles for all of the current and proposed AVAs through their [AVA Map Explorer](https://www.ttb.gov/wine/ava-map-explorer) it's probably better to create an independent repository using their dataset instead of this one. Along with some additional valuable properties provided by the UCD dataset the primary advantage it has over the data provided by TTB is the inclusion of the historical boundaries. For most use cases the TTB data is perhaps the best option, and it's official (for informational purposes only).

One of the suggestions I made to Karen Thorton at TTB was a way to retrieve all the Shapefiles without having to click on each AVA in the map and then the download link or having to request a consolidated file by email. I also suggested adding a more detailed changelog to the landing page to document which files were changed without having to download all of them each time. Perhaps your repo could be the place where they can direct users to get the information/files.

I'm very much interested in your ideas to digitize/automate the AVA map creation/approval process. I'm a fullstack Javscript developer and would be interested in helping with the project.

[Nice site!](https://robertclayvines.maps.arcgis.com/home/index.html) I really enjoyed scrolling through the [story](https://storymaps.arcgis.com/stories/0480cd9ea5cc46ba98855a3870a3fad5) and exploring the [map](https://robertclayvines.maps.arcgis.com/apps/mapviewer/index.html?webmap=17e25eac7a424b6cb2082cef6dd6091a).

Cheers!

**Comment by MicheleTobias** [comment]
The purpose of this repository is to provide a dataset of current and former AVA boundaries that is well-documented and uses methods consistent with the TTB boundary descriptions that fits the reproducibility needs of researchers. It is not intended to replace or interact with the data produced by the TTB. It is not affiliated with the TTB. At this time, we don't plan on digitizing the proposed boundaries because they are likely to change.

**Comment by MicheleTobias** [comment]
I am unclear what processes you would like to automate. We already provide individual boundary files and files containing boundaries aggregated at state and national levels. This meets the current goals of the project.
